### PR TITLE
Eric mp branch

### PR DESCRIPTION
**Pull request by GFDL-Eric**
**Description**
Created a unit test for mpp_min and mpp_max interfaces with 32-bit and 64-bit real arrays.

**How Has This Been Tested?**
Just Skylake with Intel so far... more tests welcome!

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: test_fms/mpp/test_minmax.F90 [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+program test   !test mpp_min and mpp_max functions for various ints and reals
```
Comment (ngs333) [inline test_fms/mpp/test_minmax.F90:19]: I think we are trying to use doxygen style comments for new code. Some annotations we have used are:
@author @brief @email , @description , @todo
You can see here fro more info: https://www.doxygen.nl/manual/docblocks.html#docstructure

File: test_fms/mpp/test_minmax.F90 [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  deallocate( a, a8 )
+  call MPI_FINALIZE(ierr)
+
+contains
+
+  subroutine test_mpp_max_r4
+
+  a = real(pe+1, kind=FLOAT_KIND)
+  print *, 'pe,     pe+1 =', pe, a(1)
+  call mpp_max( a(1) )
```
Comment (ngs333) [inline test_fms/mpp/test_minmax.F90:67]: cant you add an if statement to check if the value determined by mpp_max is what it should be, and if not call  cmpp_error(FATAL, " message")

File: test_fms/mpp/test_minmax.F90 [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        call mpp_set_current_pelist( (/(i,i=0,npes-2)/) )
+        id = mpp_clock_id( 'Partial mpp_sum' )
+        call mpp_clock_begin(id)
+        call mpp_sum( a(1:1000), 1000, (/(i,i=0,npes-2)/) )
+        call mpp_clock_end  (id)
+     end if
+     if( pe.EQ.root )print *, 'sum(pe+1) from 0 to npes-2=', a(1)
+     a = real(pe+1, kind=FLOAT_KIND)
+     if( pe.NE.npes-1 )call mpp_max( a(1), (/(i,i=0,npes-2)/) )
+     if( pe.EQ.root )print *, 'max(pe+1) from 0 to npes-2=', a(1)
```
Comment (ngs333) [inline test_fms/mpp/test_minmax.F90:89]: cant you add an if statement to check if the value determined by mpp_max is what it should be, and if not call mpp_error(FATAL, " message")

File: test_fms/mpp/test_minmax.F90 [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        call mpp_set_current_pelist( (/(i,i=0,npes-2)/) )
+        id = mpp_clock_id( 'Partial mpp_sum' )
+        call mpp_clock_begin(id)
+        call mpp_sum( a(1:1000), 1000, (/(i,i=0,npes-2)/) )
+        call mpp_clock_end  (id)
+     end if
+     if( pe.EQ.root )print *, 'sum(pe+1) from 0 to npes-2=', a(1)
+     a = real(pe+1, kind=FLOAT_KIND)
+     if( pe.NE.npes-1 )call mpp_min( a(1), (/(i,i=0,npes-2)/) )
+     if( pe.EQ.root )print *, 'min(pe+1) from 0 to npes-2=', a(1)
```
Comment (ngs333) [inline test_fms/mpp/test_minmax.F90:121]: can't you add an if statement to check if the value determined by mpp_max is what it should be, and if not call mpp_error(FATAL, " message") ?  Similar issues follow below.

File: test_fms/mpp/test_minmax.F90 [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+program test   !test mpp_min and mpp_max functions for various ints and reals
+#include <fms_platform.h>
```
Comment (wrongkindofdoctor) [inline test_fms/mpp/test_minmax.F90:20]: remove #include <fms_platform.h>, add use :: platform_mod, and change the type definitions.

**Review by wrongkindofdoctor — CHANGES_REQUESTED** [review]

File: test_fms/mpp/test_minmax.F90 (hunk 5 above)
Comment (GFDL-Eric) [reply test_fms/mpp/test_minmax.F90]: Updated with commit cf4c4426765b7d575b1eccbc7a11474b2fdd6f73

File: test_fms/mpp/test_minmax.F90 (hunk 1 above)
Comment (GFDL-Eric) [reply test_fms/mpp/test_minmax.F90]: Added in commit 3f3be3e9931c047098530fab532069b4629c17d4

File: test_fms/mpp/test_minmax.F90 (hunk 2 above)
Comment (GFDL-Eric) [reply test_fms/mpp/test_minmax.F90:71]: Added in commit 3f3be3e9931c047098530fab532069b4629c17d4

File: test_fms/mpp/test_minmax.F90 (hunk 3 above)
Comment (GFDL-Eric) [reply test_fms/mpp/test_minmax.F90:101]: Added in commit 3f3be3e9931c047098530fab532069b4629c17d4

File: test_fms/mpp/test_minmax.F90 (hunk 4 above)
Comment (GFDL-Eric) [reply test_fms/mpp/test_minmax.F90:141]: Added in commit 3f3be3e9931c047098530fab532069b4629c17d4

**Comment by GFDL-Eric** [comment]
Hi everyone,

I pushed a few changes here (I did a force push so I'd recommend deleting any local versions and re-pulling if you want to test).

**Comment by GFDL-Eric** [comment]
Rebased off of 6b92383

**Review by wrongkindofdoctor — APPROVED** [review]

**Pull request merged**